### PR TITLE
Upgrade to Helm Chart 3

### PR DESCRIPTION
**Pull request by orltom**
## Description
Upgrade Microgateway Chart to Helm Chart 3 and update dependencies.

## Hints
- Bitnami Redis Helm Chart Values need to be migrated. For more information see https://github.com/bitnami/charts/tree/master/bitnami/redis#upgrading
- Run unit tests with Helm3

### CODE REVIEW

File: charts/microgateway/values.yaml [hunk 1]
```diff
@@ -373,17 +373,24 @@ redis:
   # redis.enabled -- Deploy pre-configured [Redis](#redis).
   enabled: false
   fullnameOverride: redis
-  securityContext:
```
Comment (orltom) [inline charts/microgateway/values.yaml:376]: Please review the following changes carefully. With the Bitnami Redis chart upgrade, the default Redis parameters need to be mapped to the new parameters from the Redis Helm Chart 15.7.6. For more information see: https://github.com/bitnami/charts/tree/master/bitnami/redis#upgrading

File: charts/microgateway/Chart.yaml [hunk 2]
```diff
@@ -40,3 +40,12 @@ annotations:
     - name: Airlock Minikube Example
       url: https://github.com/ergon/airlock-minikube-example
   artifacthub.io/license: MIT
+dependencies:
```
Comment (orltom) [inline charts/microgateway/Chart.yaml:43]: With Helm3 the dependencies need to be declared in Chart.yaml

File: charts/microgateway/tests/__snapshot__/configmap_test.yaml.snap [hunk 3]
```diff
@@ -1,14 +0,0 @@
-dsl example:
```
Comment (orltom) [inline charts/microgateway/tests/__snapshot__/configmap_test.yaml.snap:1]: Unclear why this file is required.

File: charts/microgateway/.gitignore [hunk 4]
```diff
@@ -1 +1,3 @@
 .DS_Store
+
+tests/__snapshot__/
```
Comment (libuweber) [inline charts/microgateway/.gitignore]: This folder and the files contained within are used in the unit test configmap_test.yaml as a reference for correct output. They should not be deleted. If the file is not present, the unit test will generate a new one and pass.
The DEVELOPER_NOTES.md contains information on using / updating the snapshots.

File: charts/microgateway/Chart.yaml [hunk 5]
```diff
@@ -40,3 +40,12 @@ annotations:
...
+  - name: redis
```
Comment (libuweber) [inline charts/microgateway/Chart.yaml:44]: The dependency seems to produce a warning during installation: coalesce.go:203: warning: destination for repository is a table. Ignoring non-table value bitnami/redis

**Review by libuweber — CHANGES_REQUESTED** [review]
please fix the config map unit test

File: charts/microgateway/.gitignore (hunk 4 above)
Comment (orltom) [reply charts/microgateway/.gitignore]: Thank you for your clarification

File: charts/microgateway/Chart.yaml (hunk 5 above)
Comment (orltom) [reply charts/microgateway/Chart.yaml:44]: Unclear how to fix or ignore this warning message. Any hints are welcome :)

**Review by libuweber — APPROVED** [review]

**Pull request merged**